api/v2beta1/helmrelease_types.go

```diff
@@ -219,11 +219,32 @@ func (in HelmReleaseSpec) GetUninstall() Uninstall {
 // generate a v1beta2.HelmChart object in the same namespace as the referenced
 // v1beta2.Source.
 type HelmChartTemplate struct {
+	// ObjectMeta holds the template for metadata like labels and annotations.
+	// +optional
+	ObjectMeta HelmChartTemplateObjectMeta `json:"metadata,omitempty"`
```

Comment:
But it mirrors the (partial) of the actual objects, which is `ObjectMeta` when embedded?

Reply:
Yes, because that's also how it ends up in a Kubernetes object. The problem with `Metadata` is that it would block any growth, or e.g. later options to define the chart name directly in this part of the spec. In addition, it is not any different from: https://github.com/fluxcd/helm-controller/blob/main/api/v2beta1/helmrelease_types.go#L947

Reply:
Ok fair point!